public/app/plugins/datasource/cloudwatch/components/QueryHeader.tsx

```diff
@@ -59,7 +58,7 @@ const QueryHeader: React.FC<QueryHeaderProps> = ({
     <EditorHeader>
       <InlineSelect
         label="Region"
-        value={regions.find((v) => v.value === region)}
+        value={region}
```

Comment:
Before this change, the interpolated label was displayed in case a template variable was chosen. This change allows the field to show the actual variable, e.g `$region`.